### PR TITLE
Alerting: Prevent showing "Permissions denied" alert when not accurate

### PR DESCRIPTION
**Pull request by VikaCep**
**What is this feature?**

Users without org admin role were shown a "Permissions denied" alert in the creation/edition rule form even though they had access to create/edit the rule. The problem came after the `NotificationPreview` component made a request to `/api/v1/ngalert/alertmanagers` in order to obtain external alertmanagers information which is an endpoint that requires that permission level. 
Turns out we don't actually need to call that endpoint to obtain that information. 

**Why do we need this feature?**

To prevent showing the "Permissions denied" alert which is confusing and inaccurate.

**Who is this feature for?**

All users without org admin access.

**Which issue(s) does this PR fix?**:

Related to https://github.com/grafana/support-escalations/issues/7303

Before

After

### CODE REVIEW

**Review by soniaAguilarPeiron — APPROVED** [review]
Great!🚀

File: public/app/features/alerting/unified/components/rule-editor/notificaton-preview/useGetAlertManagersSourceNamesAndImage.tsx [hunk 1]
```diff
@@ -12,10 +11,13 @@ export const useGetAlertManagersSourceNamesAndImage = () => {
   //get current alerting config
   const { currentData: amConfigStatus } = alertmanagerApi.useGetAlertmanagerChoiceStatusQuery(undefined);
 
-  const externalDsAlertManagers: AlertManagerNameWithImage[] = useExternalDataSourceAlertmanagers().map((ds) => ({
-    name: ds.dataSource.name,
-    img: ds.dataSource.meta.info.logos.small,
-  }));
+  const externalDsAlertManagers = getAlertManagerDataSources()
```
Comment (konrad147) [inline public/app/features/alerting/unified/components/rule-editor/notificaton-preview/useGetAlertManagersSourceNamesAndImage.tsx:14]: Nice!

**Review by konrad147 — APPROVED** [review]
LGTM!

File: public/app/features/alerting/unified/components/rule-editor/notificaton-preview/useGetAlertManagersSourceNamesAndImage.tsx (hunk 1 above)
Comment (gillesdemey) [reply public/app/features/alerting/unified/components/rule-editor/notificaton-preview/useGetAlertManagersSourceNamesAndImage.tsx:14]: Can we add a function to `public/app/features/alerting/unified/utils/datasource.ts` and call it here? We might be able to use it in other places in the future if we want to circumvent the API call to `//api/v1/ngalert/alertmanagers`

**Review by gillesdemey — APPROVED** [review]
LGTM with one nit

**Pull request merged**